### ISSUE TITLE
depopts shown as [required by] in action lists

### ISSUE THREAD

**Issue by whitequark** [ocaml/opam#1814]
For example, in this case:

```
$ opam install utop
The following actions will be performed:
 - install   cppo.1.0.1                            [required by utop]
 - install   camlp4.4.01.0                         [required by utop]
 - install   react.1.2.0                           [required by utop]
 - install   camomile.0.8.5                        [required by zed]
 - recompile lwt.dev*                              [uses camlp4]
 - install   zed.1.3                               [required by lambda-term]
 - install   lambda-term.1.7                       [required by utop]
 - install   utop.1.16
=== 7 to install | 1 to reinstall ===
Do you want to continue ? [Y/n] 
```

Note that I'm using versions of utop, lambda-term and lwt that **do not** depend on camlp4. The only thing that depends on camlp4 is camomile; however utop has a depopt on camlp4, and [required by] seems to anchor to that depopt. I think [required by] should ignore depopts.

**Comment by AltGr** [comment]
Indeed: the reasons displayed are computed from the graph of actions, which includes depends and depopts, because they are treated the same for action ordering. So at that stage, it's doable, but non-trivial to add a case for optional dependencies (which may still be necessary to give an explanation for recompilations).

Also, to avoid too long listings, all actions requiring X are not shown, only the one(s) that have the shortest link to the required action: this is why it shows the direct dependency of utop (not knowing it's only a depopt) and not the dependency of camomile.

You can get a finer overview from the generated `xxx-actions.dot` file if you pass the `--cudf=xxx` option.

**Comment by whitequark** [comment]
A low-tech solution: replace `required by` with `used by`, which encompasses both depends and depopts.
